src/Microsoft.Dnx.Tooling/Publish/PublishProject.cs

```diff
@@ -375,54 +361,85 @@ private bool PrunePackages(PublishRoot root)
                 foreach (var library in target.Libraries)
                 {
                     var packagesDir = resolver.GetInstallPath(library.Name, library.Version);
-                    var manifest = resolver.GetManifestFilePath(library.Name, library.Version);
-
-                    keep.Add(manifest);
 
                     foreach (var path in library.RuntimeAssemblies)
                     {
-                        keep.Add(CombinePath(packagesDir, path));
+                        keep.Add(Path.Combine(packagesDir, Path.GetDirectoryName(path)));
```

Comment:
Instead of collecting files to keep, this collects `TFM`s. An alternative is to compare the lock files of the original project and the published output. Or some kind of fuzzy best match to figure it out from the `TFM` of the published lockfile.